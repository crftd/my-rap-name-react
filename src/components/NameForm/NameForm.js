import React, { Component } from 'react';
import { Formik, Form, Field } from 'formik';
import TextField from '@material-ui/core/TextField';
import Button from '@material-ui/core/Button';
import { MuiPickersUtilsProvider, DatePicker } from 'material-ui-pickers';
import { MuiThemeProvider, createMuiTheme } from '@material-ui/core/styles';
import DateFnsUtils from '@date-io/date-fns';
import styled from 'styled-components';

const FormGroup = styled.div`
  padding-bottom: 20px;
`;

const ButtonGroup = styled.div`
  display: flex;
  justify-content: center;
`;

const theme = createMuiTheme({
  palette: {
    primary: {
      light: '#ff9f8f',
      main: '#ff6d61',
      dark: '#c63b37',
      contrastText: '#000',
    },
    secondary: {
      light: '#ffcd50',
      main: '#f99c17',
      dark: '#c06d00',
      contrastText: '#000',
    },
  },
  typography: {
    useNextVariants: true,
  },
});

const initialValues = {
  name: '',
  birthday: new Date(1993, 2, 4),
};

const DatePickerField = ({ field, form, ...other }) => {
  const currentError = form.errors[field.name];
  return (
    <MuiPickersUtilsProvider utils={DateFnsUtils}>
      <DatePicker
        style={{ width: '100%' }}
        keyboard
        name={field.name}
        value={field.value}
        format="dd/MM/yyyy"
        helperText={currentError}
        error={Boolean(currentError)}
        onError={(_, error) => form.setFieldError(field.name, error)}
        onChange={date => form.setFieldValue(field.name, date, true)}
        mask={value =>
          value
            ? [/\d/, /\d/, '/', /\d/, /\d/, '/', /\d/, /\d/, /\d/, /\d/]
            : []
        }
        {...other}
      />
    </MuiPickersUtilsProvider>
  );
};

export default class NameForm extends Component {
  render() {
    return (
      <MuiThemeProvider theme={theme}>
        <Formik initialValues={initialValues} onSubmit={this.props.onSubmit}>
          <Form>
            <FormGroup>
              <Field name="name">
                {({ field, other }) => (
                  <TextField
                    style={{ width: '100%' }}
                    label="Name"
                    {...field}
                    {...other}
                  />
                )}
              </Field>
            </FormGroup>
            <FormGroup>
              <Field
                label="Birthday"
                name="birthday"
                component={DatePickerField}
              />
            </FormGroup>
            <ButtonGroup>
              <Button type="submit" color="primary" variant="contained">
                Get my RAP name
              </Button>
            </ButtonGroup>
          </Form>
        </Formik>
      </MuiThemeProvider>
    );
  }
}
