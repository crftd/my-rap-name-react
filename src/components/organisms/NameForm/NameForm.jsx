import React, { Component } from 'react';
import { Formik, Form, Field } from 'formik';
import Button from '@material-ui/core/Button';
import * as Yup from 'yup';

import FormGroup from '../../molecules/FormGroup/FormGroup.jsx';
import ButtonGroup from '../../molecules/ButtonGroup/ButtonGroup.jsx';
import DatePickerField from '../../atoms/DatePickerField/DatePickerField.jsx';
import TextField from '../../atoms/TextField/TextField.jsx';

const initialValues = {
  name: '',
  birthday: new Date(1993, 2, 4),
};

const nameFormSchema = Yup.object().shape({
  name: Yup.string()
    .min(2, 'Too short!')
    .required('Required'),
});

export default class NameForm extends Component {
  render() {
    return (
      <Formik
        initialValues={initialValues}
        onSubmit={this.props.onSubmit}
        validationSchema={nameFormSchema}
      >
        <Form>
          <FormGroup>
            <Field name="name" component={TextField} />
          </FormGroup>
          <FormGroup>
            <Field
              label="Birthday"
              name="birthday"
              component={DatePickerField}
            />
          </FormGroup>
          <ButtonGroup>
            <Button type="submit" color="primary" variant="contained">
              Get my RAP name
            </Button>
          </ButtonGroup>
        </Form>
      </Formik>
    );
  }
}
