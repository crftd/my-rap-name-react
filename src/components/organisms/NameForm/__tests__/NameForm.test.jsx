import React from 'react';
import { shallow } from 'enzyme';
import NameForm from '../NameForm.jsx';

describe('[Snapshot] NameForm', () => {
  it('should render', () => {
    // Arrange

    // Act
    const actualTree = shallow(<NameForm />);

    // Assert
    expect(actualTree).toMatchSnapshot();
  });
});
