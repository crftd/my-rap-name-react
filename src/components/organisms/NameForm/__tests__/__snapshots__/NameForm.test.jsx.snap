// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Snapshot] NameForm should render 1`] = `
<Formik
  enableReinitialize={false}
  initialValues={
    Object {
      "birthday": 1993-03-04T00:00:00.000Z,
      "name": "",
    }
  }
  isInitialValid={false}
  validateOnBlur={true}
  validateOnChange={true}
  validationSchema={
    ObjectSchema {
      "_blacklist": RefSet {
        "list": Set {},
        "refs": Map {},
      },
      "_conditions": Array [],
      "_defaultDefault": [Function],
      "_deps": Array [],
      "_excludedEdges": Array [],
      "_exclusive": Object {},
      "_mutate": false,
      "_nodes": Array [
        "name",
      ],
      "_options": Object {
        "abortEarly": true,
        "recursive": true,
      },
      "_type": "object",
      "_typeError": [Function],
      "_whitelist": RefSet {
        "list": Set {},
        "refs": Map {},
      },
      "fields": Object {
        "name": StringSchema {
          "_blacklist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "_conditions": Array [],
          "_deps": Array [],
          "_exclusive": Object {
            "min": true,
            "required": false,
          },
          "_mutate": false,
          "_options": Object {
            "abortEarly": true,
            "recursive": true,
          },
          "_type": "string",
          "_typeError": [Function],
          "_whitelist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "tests": Array [
            [Function],
            [Function],
            [Function],
          ],
          "transforms": Array [
            [Function],
          ],
        },
      },
      "tests": Array [],
      "transforms": Array [
        [Function],
      ],
    }
  }
>
  <Form>
    <_default>
      <FormikConnect(FieldInner)
        component={[Function]}
        name="name"
      />
    </_default>
    <_default>
      <FormikConnect(FieldInner)
        component={[Function]}
        label="Birthday"
        name="birthday"
      />
    </_default>
    <_default>
      <WithStyles(Button)
        color="primary"
        type="submit"
        variant="contained"
      >
        Get my RAP name
      </WithStyles(Button)>
    </_default>
  </Form>
</Formik>
`;
